app/services/mainstream_content_fetcher.rb

```diff
@@ -0,0 +1,12 @@
+class MainstreamContentFetcher
```

Comment:
Could you please also rename this class - it's actually fetching a subset of Mainstream Content.